lib/Sema/SemaExpr.cpp

```diff
@@ -8058,6 +8058,112 @@ Sema::CheckSingleAssignmentConstraints(QualType LHSType, ExprResult &CallerRHS,
   return result;
 }
 
+/// If a declaration has a Checked C bounds-safe interface
+/// attached to it, return the checked pointer type for the interface.
+/// Otherwise return a null QualType.
+///
+/// This falls into two cases:
+/// 1. The interface is a type annotation: return the type in the
+///    annotation.
+/// 2. The interface is a bounds expression.  This implies the checked
+/// type should be an _Array_ptr type.  Construct an _Array_ptr version
+/// of the unchecked ponter type for the declaration and return it.
+QualType Sema::GetInteropType(const ValueDecl *Decl) {
+  const DeclaratorDecl *TargetDecl = nullptr;
+  if (const FieldDecl *Field = dyn_cast<FieldDecl>(Decl))
+    TargetDecl = Field;
+  else if (const VarDecl *Var = dyn_cast<VarDecl>(Decl))
+    TargetDecl = Var;
+
+  if (TargetDecl) {
+    if (const BoundsExpr *Bounds = TargetDecl->getBoundsExpr()) {
```

Comment:
In the future, if and when we support multiple bounds expressions on a declaration--I'm assuming you'll need to take a full pass over all calls to this accessor.  Even if we don't have the concrete syntax now, might it make sense to start preparing for that eventuality in the code--so that new code you write will at least consider the possibility of multiple declarations?